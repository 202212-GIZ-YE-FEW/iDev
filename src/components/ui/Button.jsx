import clsx from "clsx";
export default function Button(prop) {
    const {
        content,
        size = "large",
        filled = "true",
        textTransform = "capitalize",
        fontSize = "xl",
        radius = "6px",
    } = prop;
    return (
        <button
            className={clsx(
                `font-normal whitespace-nowrap cursor-pointer ${textTransform} ${fontSize} rounded-${radius}`,
                {
                    "px-[10] md:px-[18px] lg:px-[28px] py-[4.7px]":
                        size === "small",
                    "px-[25px] md:px-[35px] lg:px-[59px] py-[12px] font-medium":
                        size === "large",
                    "bg-cyan text-black": filled === "true",
                    "border-2 border-cyan text-cyan": filled === "false",
                }
            )}
        >
            {content}
        </button>
    );
}

// Example of using
{
    /* <Button
    content='login'
    text-transform='uppercase'
    filled='false'
    size='large'
    fontSize="5xl"
    radius="md"
/> */
}
