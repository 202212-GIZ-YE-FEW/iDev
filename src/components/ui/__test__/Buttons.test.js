import renderer from "react-test-renderer";

import Button from "../buttons";

it("renders correctly", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
});
