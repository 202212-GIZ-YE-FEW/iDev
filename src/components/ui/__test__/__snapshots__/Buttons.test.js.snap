// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="font-normal cursor-pointer capitalize text-xl rounded-6px px-[59px] py-[12px] font-medium bg-cyan text-black"
/>
`;
